xlators/cluster/dht/src/dht-helper.c

```diff
@@ -836,7 +836,11 @@ dht_local_init(call_frame_t *frame, loc_t *loc, fd_t *fd, glusterfs_fop_t fop)
 
     if (inode) {
         local->layout = dht_layout_get(frame->this, inode);
-        local->cached_subvol = dht_subvol_get_cached(frame->this, inode);
+        if (local->layout) {
+            dht_layout_ref(local->layout);
```

Comment:
There's no need to do this. It doesn't add anything useful. We already have a reference to the layout and it's safe to use it.

Reply:
Done.